### ISSUE TITLE
More than one NeoPixel per emulated bulb/LightStip

### ISSUE THREAD

**Issue by probonopd** [probonopd/ESP8266HueEmulator#27]
Let's say I have a 30-LED NeoPixel strip but the sketch can only emulate 6 Hue LightStrips. There should be a constant that can be set to have more than one consecutive NeoPixels treated as one emulated LightStrip. In this example, I would like to set `const ledsPerBulb = 5; // treat 5 physical LEDs as one emulated bulb`.

Pull requests welcome!

**Comment by sylvainblot** [comment]
You can change it here : https://github.com/probonopd/ESP8266HueEmulator/blob/master/ESP8266HueEmulator/LightService.h#L36

**Comment by probonopd** [comment]
What I am looking for is to treat e.g., 5 physical NeoPixels as 1 logical Hue bulb...

**Comment by 2xAA** [comment]
I implemented this in a much older version about a year ago now I think, but never made a PR. The code has changed somewhat since then.

I'll check if I can reimplement this.

**Comment by probonopd** [comment]
Thanks @2xAA

**Comment by mtaram** [comment]
I cant seem to control more than 14 LEDS, I increased the buffer to 10240 to check. App gets stuck on connecting on anything over 14.

**Comment by probonopd** [comment]
@mtaram with (significantly) more than 8 _logical_ pixels we are getting memory issues. But the idea here is to combine more than 1 _physical_ into one logical pixel. In the original Hue system, light strips are treated as one logical pixel.

**Comment by mtaram** [comment]
I too was thinking towards that, unfortunately couldnt get time to experiment more. Need to read and understand how to treat the entire strip as one LED, I remember reading about it sometime back. Lets see. Will post if I make any progress.

**Comment by 2xAA** [comment]
Hi, sorry for the delay on this.
Try my fork, I've added in basically I had before.
https://github.com/2xAA/ESP8266HueEmulator

The code only extends the virtual bulbs to a set number of physical LEDs.

So, I've opted for 10 LEDs per virtual bulb. It worked well for me before when I had some cool shelves and I had 27 LEDs per three bulbs.

(photo of my old bedroom with the old PhilipsHueEmulator code running with my modifications)
![](https://scontent-lhr3-1.cdninstagram.com/t51.2885-15/e35/12728472_1548531712141613_142568271_n.jpg?ig_cache_key=MTE5MDI3NDE4MjU0OTg1Mzk3Mg%3D%3D.2)
